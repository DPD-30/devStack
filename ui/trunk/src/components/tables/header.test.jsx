import React from 'react'
import { renderToString } from 'react-dom/server'
import { HashRouter as Router } from 'react-router-dom'


import TestComponent from './header.jsx'

jest.unmock('./header.jsx')


test('TableHeader matches snapshot without subheadertext', () => {
	const comp = renderToString(
        <Router>
            <TestComponent headerText="Mock Header Text" buttonText="Mock Button Text" buttonLink="/mockURL" />
        </Router>
    )

	expect(comp).toMatchSnapshot()
})

test('TableHeader matches snapshot with subheadertext', () => {
	const comp = renderToString(
        <Router>
            <TestComponent headerText="Mock Header Text" headerSubText="And more" buttonText="Mock Button Text" buttonLink="/mockURL" />
        </Router>
    )

	expect(comp).toMatchSnapshot()
})

test('TableHeader matches snapshot with a click handler', () => {
	const comp = renderToString(
        <Router>
            <TestComponent headerText="Mock Header Text" buttonText="Mock Button Text" buttonClick={jest.fn()} />
        </Router>
    )

	expect(comp).toMatchSnapshot()
})
