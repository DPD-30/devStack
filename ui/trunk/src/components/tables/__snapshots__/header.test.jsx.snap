// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableHeader matches snapshot with a click handler 1`] = `
<div class="container-fluid">
  <div class="row">
    <div class="col-9">
      <h2>
        Mock Header Text
      </h2>
    </div>
    <div style="text-align:right"
         class="col-3"
    >
      <button type="button"
              class="btn btn-primary"
      >
        Mock Button Text
      </button>
    </div>
  </div>
</div>
`;

exports[`TableHeader matches snapshot with subheadertext 1`] = `
<div class="container-fluid">
  <div class="row">
    <div class="col-9">
      <h2>
        Mock Header Text
      </h2>
      <h3>
        And more
      </h3>
    </div>
    <div style="text-align:right"
         class="col-3"
    >
      <a href="#/mockURL">
        <button class="btn btn-primary">
          Mock Button Text
        </button>
      </a>
    </div>
  </div>
</div>
`;

exports[`TableHeader matches snapshot without subheadertext 1`] = `
<div class="container-fluid">
  <div class="row">
    <div class="col-9">
      <h2>
        Mock Header Text
      </h2>
    </div>
    <div style="text-align:right"
         class="col-3"
    >
      <a href="#/mockURL">
        <button class="btn btn-primary">
          Mock Button Text
        </button>
      </a>
    </div>
  </div>
</div>
`;
